sql/rowexec: TestUncertaintyErrorIsReturned failed

sql/rowexec.TestUncertaintyErrorIsReturned [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11205084?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11205084?buildTab=artifacts#/) on master @ [695d4ec85157a4da0c97237e6f16bf919eebf6c5](https://github.com/cockroachdb/cockroach/commits/695d4ec85157a4da0c97237e6f16bf919eebf6c5):


```
        	            				github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:771
        	            				github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:808
        	Error:      	Not equal: 
        	            	expected: "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJy8lW9P4koUxt_fTzE5yY16M0inFPA2McEIN-JFcIFkNzHEjPQAjaXDzkwjxvDdN21xpYV2Kf7pC2WY098855knhxdQPz2wYdDqtC6HxPUngvzX792Qu9aP285Fu0uOm-3BcPCtQ8ng6uK2dULWpf_EdZpc99pdInyU4on0ukSfLsn5en26HJHvV61-KwZ32v-3yFHT5VPJ5_bfR0DBFw52-RwV2HfAgIIJFCoworCQYoxKCRluvUSFbWcJtkHB9ReBDr8eURgLiWC_gHa1h2DDkD942EfuoCwbQMFBzV0vwuuGvl884jNQuBReMPeVTZaUhOvBgoerUpkZMFpREIFeH_FGfngmM65mSWaDwWg1oqA0nyLYbEN3uwm2saIZ0t-4gS-kgxKdBHkUvvmnkh39X3E1uxauj7JsJaV6ONHHDXZyLt3pLPoEFHqBtkmD0YZJG5VU629tVVJtWe9oa4fmriiJRbme7n-nFCslpZ6QwvYPBysajjIzSmXzY_PBMtV_QT6qn5aPaqItc_9LMQtfimmUPvRGzEOlV5LHxAOwEf_bbmJj5rxLvpmSX8mU_wWBqn1aoGqZA2eHoj6qhfAV7jVPjNRJJRY2ic4UY9OUCOQYb6UYR7XxsheBovA4qHS8u160_dctpSXy-e-fgf1JVjbJKkaqZ5PO0iSWJhmbJDObxMw0ysxFnSVQRq5RlUMtZ8VIOZZXi5FyLP83TbIOtrySRlUPNWrr8vJJOUbVipFyjGJbOagVaM_cRG0ZlU-yskn1YqR6NolthbN-cBCscFxNPPF07zpgg7F-Sjv-vD4QvsCnKpyZg5l4irjD50U48SbcU0jhhj9iEzXKueu7SrtjsLUMcLX661cAAAD__3vS9aQ="
        	            	actual  : "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJycklFr2zAUhd_3K8SF0XYoreW8CQoujUfcpUkWBzYopmjxTSLqSJ4kk5SQ_z5kZ2vipds6Pdi-0tV3zzl4C_Z7ARzSeBDfTolUc00-Tkb35CH-Oh7cJENy3kvSafp5QEnavxnHF2Tf-qHpc-RulAyJVmj0moyGxF1uyPW-vtxk5Es_nsQNeJB8islZT4qFESv-_gwoKJ3jUKzQAn8ABhmF0ugZWquN39rWDUm-AR5QkKqsnN_OKMy0QeBbcNIVCBym4luBExQ5mqsAKOTohCxqrIvcY_mEz0DhVhfVSllONpT4Oi2FrzqQ7Sjoyr3wrRMLBM4OBCU94MGO_p8mdqypySdqXr-r-xdpYUsae1Xai6JKaZOjwfxITeZv_q3lhL--sMs7LRWaq_DYXoFzdx6xi2sjF8v6CyiMKsdJxGgU0qj7qq1uy1b4lsQnaEutLLbtnZwUtCZ1mDeJ-QKb0KyuzAzHRs_q3qYc1aB6I0frmtOwKRJVHzE_waBY_fphDknsDaTwkMTapPCPpO4RKTjWlFGYF3r9KHPgEOxX58Tj5wJ_QSysDztd6nWNnT6XPqq5KCxSuBdP2EOHZiWVtE7OgDtT4W737kcAAAD__1-EaU0="
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJy8lW9P4koUxt_fTzE5yY16M0inFPA2McEIN-JFcIFkNzHEjPQAjaXDzkwjxvDdN21xpYV2Kf7pC2WY098855knhxdQPz2wYdDqtC6HxPUngvzX792Qu9aP285Fu0uOm-3BcPCtQ8ng6uK2dULWpf_EdZpc99pdInyU4on0ukSfLsn5en26HJHvV61-KwZ32v-3yFHT5VPJ5_bfR0DBFw52-RwV2HfAgIIJFCoworCQYoxKCRluvUSFbWcJtkHB9ReBDr8eURgLiWC_gHa1h2DDkD942EfuoCwbQMFBzV0vwuuGvl884jNQuBReMPeVTZaUhOvBgoerUpkZMFpREIFeH_FGfngmM65mSWaDwWg1oqA0nyLYbEN3uwm2saIZ0t-4gS-kgxKdBHkUvvmnkh39X3E1uxauj7JsJaV6ONHHDXZyLt3pLPoEFHqBtkmD0YZJG5VU629tVVJtWe9oa4fmriiJRbme7n-nFCslpZ6QwvYPBysajjIzSmXzY_PBMtV_QT6qn5aPaqItc_9LMQtfimmUPvRGzEOlV5LHxAOwEf_bbmJj5rxLvpmSX8mU_wWBqn1aoGqZA2eHoj6qhfAV7jVPjNRJJRY2ic4UY9OUCOQYb6UYR7XxsheBovA4qHS8u160_dctpSXy-e-fgf1JVjbJKkaqZ5PO0iSWJhmbJDObxMw0ysxFnSVQRq5RlUMtZ8VIOZZXi5FyLP83TbIOtrySRlUPNWrr8vJJOUbVipFyjGJbOagVaM_cRG0ZlU-yskn1YqR6NolthbN-cBCscFxNPPF07zpgg7F-Sjv-vD4QvsCnKpyZg5l4irjD50U48SbcU0jhhj9iEzXKueu7SrtjsLUMcLX661cAAAD__3vS9aQ=
        	            	+Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJycklFr2zAUhd_3K8SF0XYoreW8CQoujUfcpUkWBzYopmjxTSLqSJ4kk5SQ_z5kZ2vipds6Pdi-0tV3zzl4C_Z7ARzSeBDfTolUc00-Tkb35CH-Oh7cJENy3kvSafp5QEnavxnHF2Tf-qHpc-RulAyJVmj0moyGxF1uyPW-vtxk5Es_nsQNeJB8islZT4qFESv-_gwoKJ3jUKzQAn8ABhmF0ugZWquN39rWDUm-AR5QkKqsnN_OKMy0QeBbcNIVCBym4luBExQ5mqsAKOTohCxqrIvcY_mEz0DhVhfVSllONpT4Oi2FrzqQ7Sjoyr3wrRMLBM4OBCU94MGO_p8mdqypySdqXr-r-xdpYUsae1Xai6JKaZOjwfxITeZv_q3lhL--sMs7LRWaq_DYXoFzdx6xi2sjF8v6CyiMKsdJxGgU0qj7qq1uy1b4lsQnaEutLLbtnZwUtCZ1mDeJ-QKb0KyuzAzHRs_q3qYc1aB6I0frmtOwKRJVHzE_waBY_fphDknsDaTwkMTapPCPpO4RKTjWlFGYF3r9KHPgEOxX58Tj5wJ_QSysDztd6nWNnT6XPqq5KCxSuBdP2EOHZiWVtE7OgDtT4W737kcAAAD__1-EaU0=
        	Test:       	TestUncertaintyErrorIsReturned/vectorize=false/SELECT_*_FROM_t_JOIN_onerow_ON_t.x_=_onerow.x
        --- FAIL: TestUncertaintyErrorIsReturned/vectorize=false/SELECT_*_FROM_t_JOIN_onerow_ON_t.x_=_onerow.x (0.01s)
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true
    --- FAIL: TestUncertaintyErrorIsReturned/vectorize=true (0.03s)
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_INNER_LOOKUP_JOIN_t_AS_t2_ON_t1.x_=_t2.y
    processors_test.go:770: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:770
        	            				github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:771
        	            				github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:808
        	Error:      	Not equal: 
        	            	expected: "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJzElV1v2jwUgO_fX2Ed6VW3yZTY4TPSJKqSqWlpwgjTJlWoyojLvIY4sx0VVPHfpwBdIW0iwkWbCyTHh8ePfY5PHkH9icAC3x7Y52PE4zuBvoy8a3Rj_xgOzhwXfeg7_tj_OsDIvzgb2h_RNvTTJk6jMx9pghzXtUdo4HlX34bo0nPc7QxFnos0OV2gz0jT0-UEfb-wR_ZmpYFzZaOTPg9mMphb_58AhliEzA3mTIF1AwQwUMBgwgRDIsWUKSVkNvW4DnTCBVgGBh4nqc5eTzBMhWRgPYLmOmJgwTj4GbERC0Im6wZgCJkOeLTG656-Te7ZEjCciyidx8pCC4yysZ8E2ahWJwZMVhhEqp-XUDqYMbDIjpPTB8tY4cO1LgWPt1bmS6vlLQ8XgGEgxH2aoN-Cx0jEFuqRXdclRlI88LDQkOYMzSMNW4Xn9kKwsXeYgMFLdeaNexT3mrhnFsqaOdlWoeyzYxoLGTLJwj3ByeqV7biiJpJ6Nxf4ukojp9LdUyGHFxypWnB1YtTq9PCaI1XMdjLaeLOaaxxp2H6PmmvvydLDE00rJ5oatYOzTKto7Zxh882y3DzSsPMeWe5UaYMjphIRK3ZQ4zByK9VI1opYOGObvqVEKqdsKMV0HbsZemvQ-kXIlN7Mks3AiZ-mlJYsmP_7zuySSCmJFpPMPImWksw9EtkltfIks3x3RoXtNUpRzWISyZOapaRWMamRJ7WOPah2ntQuJXWKnWie1CkldYtJzType-zuOlm530Xi4ZaHYIGxfWqv_Dw9kP0hmKnszvm_xMMaO14m2Y25CyLFMFwH96zPNJNzHnOl-RQsLVO2Wv33NwAA___OinSA"
        	            	actual  : "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyckm-L2kwQwN8_n2IYeLi27F3d2L5ZKChnyuUul1hjaeEIsjWj3Rqz6e4GFfG7lyRe63lV2u6LwPzJb34Ms0X7PUeBiR_612NQxUzD-1F8Dw_-52HYDyJ4MQiScfIhZJDc9If-S9i3vmr7HPQTcByCKPJHEMbx3cch3MZBtK94EEfg-NUa3oHzrjYpfLrxR347KQzufLgYKDk3cin-v0CGhc4okkuyKB6QY8qwNHpK1mpTp7ZNQ5CtUXQYqqKsXJ1OGU61IRRbdMrlhALH8ktOI5IZmdcdZJiRkypvsK7nJuWCNsjwWufVsrAC1gzqOCllHV1iumOoK_eLb52cEwp-IBQMUHR27M-dbrUq9kr8udJmorI1Mgy1XlQlfNOqAF0I6PFD0Q0Do1cqO2noHRnyfzT0Ti7tmeCbJ5tEhnHlam_W81jvLet1T8p2j2S9v5EdkS11YemJ6KlJnaNJl3yXMqRsTu1dWV2ZKQ2Nnja9bRg3oCaRkXVtlbdBUDyWrDMklz-v4ZDEz5K80yR-TPLOkrrnnFKGs1yvJipDgZ39u_zN5_Fh_YOc23rZyVe9arDjTVmvaiZzSwzv5YIG5MgsVaGsU1MUzlS02_33IwAA__-buF6V"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJzElV1v2jwUgO_fX2Ed6VW3yZTY4TPSJKqSqWlpwgjTJlWoyojLvIY4sx0VVPHfpwBdIW0iwkWbCyTHh8ePfY5PHkH9icAC3x7Y52PE4zuBvoy8a3Rj_xgOzhwXfeg7_tj_OsDIvzgb2h_RNvTTJk6jMx9pghzXtUdo4HlX34bo0nPc7QxFnos0OV2gz0jT0-UEfb-wR_ZmpYFzZaOTPg9mMphb_58AhliEzA3mTIF1AwQwUMBgwgRDIsWUKSVkNvW4DnTCBVgGBh4nqc5eTzBMhWRgPYLmOmJgwTj4GbERC0Im6wZgCJkOeLTG656-Te7ZEjCciyidx8pCC4yysZ8E2ahWJwZMVhhEqp-XUDqYMbDIjpPTB8tY4cO1LgWPt1bmS6vlLQ8XgGEgxH2aoN-Cx0jEFuqRXdclRlI88LDQkOYMzSMNW4Xn9kKwsXeYgMFLdeaNexT3mrhnFsqaOdlWoeyzYxoLGTLJwj3ByeqV7biiJpJ6Nxf4ukojp9LdUyGHFxypWnB1YtTq9PCaI1XMdjLaeLOaaxxp2H6PmmvvydLDE00rJ5oatYOzTKto7Zxh882y3DzSsPMeWe5UaYMjphIRK3ZQ4zByK9VI1opYOGObvqVEKqdsKMV0HbsZemvQ-kXIlN7Mks3AiZ-mlJYsmP_7zuySSCmJFpPMPImWksw9EtkltfIks3x3RoXtNUpRzWISyZOapaRWMamRJ7WOPah2ntQuJXWKnWie1CkldYtJzType-zuOlm530Xi4ZaHYIGxfWqv_Dw9kP0hmKnszvm_xMMaO14m2Y25CyLFMFwH96zPNJNzHnOl-RQsLVO2Wv33NwAA___OinSA
        	            	+Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyckm-L2kwQwN8_n2IYeLi27F3d2L5ZKChnyuUul1hjaeEIsjWj3Rqz6e4GFfG7lyRe63lV2u6LwPzJb34Ms0X7PUeBiR_612NQxUzD-1F8Dw_-52HYDyJ4MQiScfIhZJDc9If-S9i3vmr7HPQTcByCKPJHEMbx3cch3MZBtK94EEfg-NUa3oHzrjYpfLrxR347KQzufLgYKDk3cin-v0CGhc4okkuyKB6QY8qwNHpK1mpTp7ZNQ5CtUXQYqqKsXJ1OGU61IRRbdMrlhALH8ktOI5IZmdcdZJiRkypvsK7nJuWCNsjwWufVsrAC1gzqOCllHV1iumOoK_eLb52cEwp-IBQMUHR27M-dbrUq9kr8udJmorI1Mgy1XlQlfNOqAF0I6PFD0Q0Do1cqO2noHRnyfzT0Ti7tmeCbJ5tEhnHlam_W81jvLet1T8p2j2S9v5EdkS11YemJ6KlJnaNJl3yXMqRsTu1dWV2ZKQ2Nnja9bRg3oCaRkXVtlbdBUDyWrDMklz-v4ZDEz5K80yR-TPLOkrrnnFKGs1yvJipDgZ39u_zN5_Fh_YOc23rZyVe9arDjTVmvaiZzSwzv5YIG5MgsVaGsU1MUzlS02_33IwAA__-buF6V
        	Test:       	TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_INNER_LOOKUP_JOIN_t_AS_t2_ON_t1.x_=_t2.y
        --- FAIL: TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_INNER_LOOKUP_JOIN_t_AS_t2_ON_t1.x_=_t2.y (0.01s)
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x
    processors_test.go:770: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:770
        	            				github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:771
        	            				github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors_test.go:808
        	Error:      	Not equal: 
        	            	expected: "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJzEld9u2jAUh-_3FNaRpraTKbHD30iVqApT6Sh0gLRJFapcYiBqiJltVKqKd58CdC0J8UjWslwgTJwvv2N_5jyD-uWDA71Gq3HRR14wEuhrt3ONbhs_b1rnzTY6rjd7_d73Fka9y_ObxgnaTP2ynqfReQ9pgq46zfZmQFGnjTQ5XaAzpOnpYoB-XDa6jTW81fzWQEd1j40lmzqfjwBDIFzeZlOuwLkFAhgoYLBhgGEmxZArJWR463k1sekuwLEweMFsrsOfBxiGQnJwnkF72ufgQJ_d-7zLmctl3gIMLtfM81d4XdN3swf-BBguhD-fBspBC4zCcW_GwlEuTywYLDGIud684pV8_4QmTE22mTUCg-UAg9JszMEhb3I36-BYS5wtun3w6DQS3U6M_sqdB0K6XHJ3izwIn_zblB31XzI1uRJewGW-tB3V5yN9XCMnZ9IbT1bfAENnrh1UI7hGcc3GtUKk-tfK7EhlpX-obEfstsiJWb4aXYKdUQqRKNWtKGR_P0haP_LEyuXp-9pNsqYv_I_0UcELiekPIHj5IwUvb1VG998XmnpfqJV7V6Vo1ujFg0eP-lRMjH4Anyof6VMlTRfrcjUTgeJ7_R9akTflSFgnd8d8vW5KzOWQ30gxXM1dDzsr0Modlyu9vkvXg2bwcktpydn0TxPen1RKJpXSkarJJEKjKJKiPPoWZacjlZJJ1XSkajKJFKMoGkVZb1G2YaUqUZRtRBFri2UZTShkdYqkIxmcKqcjmZyKmVDM6lQhHcngFImtuRllkip2_EqZpYqpXs5qQuwgm0kGE2Kim0kmE2L7V8lqQuwgm0kmE2JnxowymRA7NNWsJlAr7DgjXzzeeS44YG2u3I6PlwvCB9hYhW2vNxGPK27_aRY2rRHzFcdwzR54nWsup17gKe0NwdFyzpfLT78DAAD__5SzAcM="
        	            	actual  : "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJysktFr2zAQxt_3V4iD0XYoreXsSVBwaDziLk2yOLBBMUWLL46obXmSTFNC_vchO1sTk250TA8hd_r0u-87vAXzIwcOcTgObxZElitFPs2nd-Q-_DYbD6IJOR9G8SL-MqYkHg1m4QXZSz-0OksGMbGM3E6jyb7wyXRCLLvckGti_ctNQr6OwnnYwsfR55CcDaXItCj4-zOgUKoUJ6JAA_weGCQUKq2WaIzSrrVtBFG6Ae5RkGVVW9dOKCyVRuBbsNLmCBwW4nuOcxQp6isPKKRohcwbrA3sQ_WIz0DhRuV1URpONpS4Oq6Eq3qQ7Cio2r7wjRUZAmcHhqIhcG9H_80T-3-e_I4n9qqnFyt1qXSKGtMjG4l7-TfJiWAjYda3Spaor_zjXDmu7HnALq61zNbNP6AwrS0nAaOBT4M-DT6-mqzfSea_ZdtzNJUqDXYTnpzkdSb1mMuJaYbt3oyq9RJnWi0bbVtOG1DTSNHY9tZvi6hsrpiboFEUvz-WQxJ7A8k_JLEuyf8jqX9E8o49JRRWuXp6kClw8Pand-Ln1wH3QGTGLTteq6cGu3iu3KpWIjdI4U484hAt6kKW0li5BG51jbvdu58BAAD__-b5ZXA="
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJzEld9u2jAUh-_3FNaRpraTKbHD30iVqApT6Sh0gLRJFapcYiBqiJltVKqKd58CdC0J8UjWslwgTJwvv2N_5jyD-uWDA71Gq3HRR14wEuhrt3ONbhs_b1rnzTY6rjd7_d73Fka9y_ObxgnaTP2ynqfReQ9pgq46zfZmQFGnjTQ5XaAzpOnpYoB-XDa6jTW81fzWQEd1j40lmzqfjwBDIFzeZlOuwLkFAhgoYLBhgGEmxZArJWR463k1sekuwLEweMFsrsOfBxiGQnJwnkF72ufgQJ_d-7zLmctl3gIMLtfM81d4XdN3swf-BBguhD-fBspBC4zCcW_GwlEuTywYLDGIud684pV8_4QmTE22mTUCg-UAg9JszMEhb3I36-BYS5wtun3w6DQS3U6M_sqdB0K6XHJ3izwIn_zblB31XzI1uRJewGW-tB3V5yN9XCMnZ9IbT1bfAENnrh1UI7hGcc3GtUKk-tfK7EhlpX-obEfstsiJWb4aXYKdUQqRKNWtKGR_P0haP_LEyuXp-9pNsqYv_I_0UcELiekPIHj5IwUvb1VG998XmnpfqJV7V6Vo1ujFg0eP-lRMjH4Anyof6VMlTRfrcjUTgeJ7_R9akTflSFgnd8d8vW5KzOWQ30gxXM1dDzsr0Modlyu9vkvXg2bwcktpydn0TxPen1RKJpXSkarJJEKjKJKiPPoWZacjlZJJ1XSkajKJFKMoGkVZb1G2YaUqUZRtRBFri2UZTShkdYqkIxmcKqcjmZyKmVDM6lQhHcngFImtuRllkip2_EqZpYqpXs5qQuwgm0kGE2Kim0kmE2L7V8lqQuwgm0kmE2JnxowymRA7NNWsJlAr7DgjXzzeeS44YG2u3I6PlwvCB9hYhW2vNxGPK27_aRY2rRHzFcdwzR54nWsup17gKe0NwdFyzpfLT78DAAD__5SzAcM=
        	            	+Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJysktFr2zAQxt_3V4iD0XYoreXsSVBwaDziLk2yOLBBMUWLL46obXmSTFNC_vchO1sTk250TA8hd_r0u-87vAXzIwcOcTgObxZElitFPs2nd-Q-_DYbD6IJOR9G8SL-MqYkHg1m4QXZSz-0OksGMbGM3E6jyb7wyXRCLLvckGti_ctNQr6OwnnYwsfR55CcDaXItCj4-zOgUKoUJ6JAA_weGCQUKq2WaIzSrrVtBFG6Ae5RkGVVW9dOKCyVRuBbsNLmCBwW4nuOcxQp6isPKKRohcwbrA3sQ_WIz0DhRuV1URpONpS4Oq6Eq3qQ7Cio2r7wjRUZAmcHhqIhcG9H_80T-3-e_I4n9qqnFyt1qXSKGtMjG4l7-TfJiWAjYda3Spaor_zjXDmu7HnALq61zNbNP6AwrS0nAaOBT4M-DT6-mqzfSea_ZdtzNJUqDXYTnpzkdSb1mMuJaYbt3oyq9RJnWi0bbVtOG1DTSNHY9tZvi6hsrpiboFEUvz-WQxJ7A8k_JLEuyf8jqX9E8o49JRRWuXp6kClw8Pand-Ln1wH3QGTGLTteq6cGu3iu3KpWIjdI4U484hAt6kKW0li5BG51jbvdu58BAAD__-b5ZXA=
        	Test:       	TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x
        --- FAIL: TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUncertaintyErrorIsReturned.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30385

--- a/pkg/sql/rowexec/aggregator_test.go
+++ b/pkg/sql/rowexec/aggregator_test.go
@@ -60,6 +60,7 @@ func aggregations(aggTestSpecs []aggTestSpec) []execinfrapb.AggregatorSpec_Aggre
 //	VARIANCE
 func TestAggregator(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	var (
 		col0              = []uint32{0}

--- a/pkg/sql/rowexec/backfiller_test.go
+++ b/pkg/sql/rowexec/backfiller_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/errors"
 	"github.com/stretchr/testify/require"
@@ -65,6 +66,7 @@ func TestingWriteResumeSpan(
 
 func TestWriteResumeSpan(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
 

--- a/pkg/sql/rowexec/distinct_test.go
+++ b/pkg/sql/rowexec/distinct_test.go
@@ -30,6 +30,7 @@ import (
 
 func TestDistinct(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [15]rowenc.EncDatum{}
 	for i := range v {

--- a/pkg/sql/rowexec/filterer_test.go
+++ b/pkg/sql/rowexec/filterer_test.go
@@ -30,6 +30,8 @@ import (
 
 func TestFilterer(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	v := [10]rowenc.EncDatum{}
 	for i := range v {
 		v[i] = rowenc.DatumToEncDatum(types.Int, tree.NewDInt(tree.DInt(i)))

--- a/pkg/sql/rowexec/hashjoiner_test.go
+++ b/pkg/sql/rowexec/hashjoiner_test.go
@@ -991,6 +991,7 @@ func mirrorJoinTypeAndOnExpr(
 
 func TestHashJoiner(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	testCases := hashJoinerTestCases()
 
@@ -1101,6 +1102,7 @@ func TestHashJoiner(t *testing.T) {
 
 func TestHashJoinerError(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [10]rowenc.EncDatum{}
 	for i := range v {
@@ -1211,6 +1213,8 @@ func checkExpectedRows(
 // the consumer is draining.
 func TestHashJoinerDrain(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	v := [10]rowenc.EncDatum{}
 	for i := range v {
 		v[i] = rowenc.DatumToEncDatum(types.Int, tree.NewDInt(tree.DInt(i)))
@@ -1319,6 +1323,7 @@ func TestHashJoinerDrain(t *testing.T) {
 // joiner will drain both inputs.
 func TestHashJoinerDrainAfterBuildPhaseError(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [10]rowenc.EncDatum{}
 	for i := range v {

--- a/pkg/sql/rowexec/inverted_expr_evaluator_test.go
+++ b/pkg/sql/rowexec/inverted_expr_evaluator_test.go
@@ -18,6 +18,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/sql/inverted"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/stretchr/testify/require"
 )
 
@@ -89,6 +90,7 @@ func keyIndexesToString(indexes [][]KeyIndex) string {
 
 func TestSetContainerUnion(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	type testCase struct {
 		a        setContainer
@@ -111,6 +113,7 @@ func TestSetContainerUnion(t *testing.T) {
 
 func TestSetContainerIntersection(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	type testCase struct {
 		a        setContainer
@@ -139,6 +142,7 @@ type keyAndIndex struct {
 // Tests both invertedExprEvaluator and batchedInvertedExprEvaluator.
 func TestInvertedExpressionEvaluator(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	leaf1 := &spanExpression{
 		FactoredUnionSpans: []invertedSpan{{Start: []byte("a"), End: []byte("d")}},
@@ -302,6 +306,7 @@ func TestInvertedExpressionEvaluator(t *testing.T) {
 // overlapping spans.
 func TestFragmentedSpans(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	expr1 := inverted.SpanExpressionProto{
 		Node: spanExpression{
@@ -362,6 +367,8 @@ func (t *testPreFilterer) PreFilter(
 
 func TestInvertedExpressionEvaluatorPreFilter(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	// Setup expressions such that the same expression appears multiple times
 	// in a span.
 	leaf1 := &spanExpression{

--- a/pkg/sql/rowexec/inverted_filterer_test.go
+++ b/pkg/sql/rowexec/inverted_filterer_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 )
 
 func intToEncodedInvertedVal(v int64) []byte {
@@ -38,6 +39,7 @@ func intSpanToEncodedSpan(start, end int64) inverted.SpanExpressionProto_Span {
 
 func TestInvertedFilterer(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	// Tests do simple intersection and reordering of columns, to exercise the
 	// plumbing in invertedFilterer -- all the heavy lifting for filtering is

--- a/pkg/sql/rowexec/inverted_joiner_test.go
+++ b/pkg/sql/rowexec/inverted_joiner_test.go
@@ -38,6 +38,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/json"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/cockroachdb/errors"
@@ -199,6 +200,8 @@ func (jsonUnionExpr) PreFilter(_ inverted.EncVal, _ []interface{}, _ []bool) (bo
 
 func TestInvertedJoiner(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	ctx := context.Background()
 
 	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
@@ -742,6 +745,7 @@ func TestInvertedJoiner(t *testing.T) {
 
 func TestInvertedJoinerDrain(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(context.Background())

--- a/pkg/sql/rowexec/mergejoiner_test.go
+++ b/pkg/sql/rowexec/mergejoiner_test.go
@@ -45,6 +45,7 @@ type mergeJoinerTestCase struct {
 
 func TestMergeJoiner(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [10]rowenc.EncDatum{}
 	for i := range v {
@@ -761,6 +762,7 @@ func TestMergeJoiner(t *testing.T) {
 // Test that the joiner shuts down fine if the consumer is closed prematurely.
 func TestConsumerClosed(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [10]rowenc.EncDatum{}
 	for i := range v {

--- a/pkg/sql/rowexec/ordinality_test.go
+++ b/pkg/sql/rowexec/ordinality_test.go
@@ -30,6 +30,7 @@ import (
 
 func TestOrdinality(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [15]rowenc.EncDatum{}
 	for i := range v {

--- a/pkg/sql/rowexec/processors_test.go
+++ b/pkg/sql/rowexec/processors_test.go
@@ -41,13 +41,13 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/distsqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/randutil"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/errors"
 	"github.com/jackc/pgx/v4"
 	"github.com/stretchr/testify/require"
 )
@@ -714,8 +714,6 @@ func TestUncertaintyErrorIsReturned(t *testing.T) {
 		// The default behavior is to enable uncertainty errors for a single random
 		// node.
 		overrideErrorOrigin []int
-		// if non-empty, this test will be skipped.
-		skip string
 	}{
 		{
 			query:           "SELECT * FROM t AS t1 JOIN t AS t2 ON t1.x = t2.x",
@@ -726,8 +724,8 @@ func TestUncertaintyErrorIsReturned(t *testing.T) {
 			expectedPlanURL: "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJzElV1v2jwUgO_fX2Ed6VW3yZTY4TPSJKqSqWlpwgjTJlWoyojLvIY4sx0VVPHfpwBdIW0iwkWbCyTHh8ePfY5PHkH9icAC3x7Y52PE4zuBvoy8a3Rj_xgOzhwXfeg7_tj_OsDIvzgb2h_RNvTTJk6jMx9pghzXtUdo4HlX34bo0nPc7QxFnos0OV2gz0jT0-UEfb-wR_ZmpYFzZaOTPg9mMphb_58AhliEzA3mTIF1AwQwUMBgwgRDIsWUKSVkNvW4DnTCBVgGBh4nqc5eTzBMhWRgPYLmOmJgwTj4GbERC0Im6wZgCJkOeLTG656-Te7ZEjCciyidx8pCC4yysZ8E2ahWJwZMVhhEqp-XUDqYMbDIjpPTB8tY4cO1LgWPt1bmS6vlLQ8XgGEgxH2aoN-Cx0jEFuqRXdclRlI88LDQkOYMzSMNW4Xn9kKwsXeYgMFLdeaNexT3mrhnFsqaOdlWoeyzYxoLGTLJwj3ByeqV7biiJpJ6Nxf4ukojp9LdUyGHFxypWnB1YtTq9PCaI1XMdjLaeLOaaxxp2H6PmmvvydLDE00rJ5oatYOzTKto7Zxh882y3DzSsPMeWe5UaYMjphIRK3ZQ4zByK9VI1opYOGObvqVEKqdsKMV0HbsZemvQ-kXIlN7Mks3AiZ-mlJYsmP_7zuySSCmJFpPMPImWksw9EtkltfIks3x3RoXtNUpRzWISyZOapaRWMamRJ7WOPah2ntQuJXWKnWie1CkldYtJzType-zuOlm530Xi4ZaHYIGxfWqv_Dw9kP0hmKnszvm_xMMaO14m2Y25CyLFMFwH96zPNJNzHnOl-RQsLVO2Wv33NwAA___OinSA",
 		},
 		{
-			// This test reproduces 51458 and should be enabled once that issue is
-			// fixed.
+			// Reproduction of not propagating errors to all outputs of the
+			// hash router (#51458).
 			query:               "SELECT * FROM t JOIN onerow ON t.x = onerow.x",
 			expectedPlanURL:     "Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJy8lW9P4koUxt_fTzE5yY16M0inFPA2McEIN-JFcIFkNzHEjPQAjaXDzkwjxvDdN21xpYV2Kf7pC2WY098855knhxdQPz2wYdDqtC6HxPUngvzX792Qu9aP285Fu0uOm-3BcPCtQ8ng6uK2dULWpf_EdZpc99pdInyU4on0ukSfLsn5en26HJHvV61-KwZ32v-3yFHT5VPJ5_bfR0DBFw52-RwV2HfAgIIJFCoworCQYoxKCRluvUSFbWcJtkHB9ReBDr8eURgLiWC_gHa1h2DDkD942EfuoCwbQMFBzV0vwuuGvl884jNQuBReMPeVTZaUhOvBgoerUpkZMFpREIFeH_FGfngmM65mSWaDwWg1oqA0nyLYbEN3uwm2saIZ0t-4gS-kgxKdBHkUvvmnkh39X3E1uxauj7JsJaV6ONHHDXZyLt3pLPoEFHqBtkmD0YZJG5VU629tVVJtWe9oa4fmriiJRbme7n-nFCslpZ6QwvYPBysajjIzSmXzY_PBMtV_QT6qn5aPaqItc_9LMQtfimmUPvRGzEOlV5LHxAOwEf_bbmJj5rxLvpmSX8mU_wWBqn1aoGqZA2eHoj6qhfAV7jVPjNRJJRY2ic4UY9OUCOQYb6UYR7XxsheBovA4qHS8u160_dctpSXy-e-fgf1JVjbJKkaqZ5PO0iSWJhmbJDObxMw0ysxFnSVQRq5RlUMtZ8VIOZZXi5FyLP83TbIOtrySRlUPNWrr8vJJOUbVipFyjGJbOagVaM_cRG0ZlU-yskn1YqR6NolthbN-cBCscFxNPPF07zpgg7F-Sjv-vD4QvsCnKpyZg5l4irjD50U48SbcU0jhhj9iEzXKueu7SrtjsLUMcLX661cAAAD__3vS9aQ=",
 			overrideErrorOrigin: []int{0},
@@ -754,14 +752,13 @@ func TestUncertaintyErrorIsReturned(t *testing.T) {
 		}
 		for _, testCase := range testCases {
 			t.Run(testCase.query, func(t *testing.T) {
-				if testCase.skip != "" {
-					skip.IgnoreLint(t, testCase.skip)
-				}
 				func() {
 					_, err := defaultConn.Exec(ctx, fmt.Sprintf("set vectorize=%s", vectorizeOpt))
 					require.NoError(t, err)
-					func() {
-						// Check distsql plan.
+					// We allow for the DistSQL plan to be different for some
+					// time in case the range cache wasn't populated as we
+					// expected (we've seen this under race in #108250).
+					testutils.SucceedsSoon(t, func() error {
 						rows, err := defaultConn.Query(
 							ctx,
 							fmt.Sprintf("SELECT info FROM [EXPLAIN (DISTSQL, SHAPE) %s] WHERE info LIKE 'Diagram:%%'", testCase.query),
@@ -771,8 +768,14 @@ func TestUncertaintyErrorIsReturned(t *testing.T) {
 						rows.Next()
 						var actualPlanURL string
 						require.NoError(t, rows.Scan(&actualPlanURL))
-						require.Equal(t, testCase.expectedPlanURL, actualPlanURL)
-					}()
+						if testCase.expectedPlanURL != actualPlanURL {
+							return errors.Newf(
+								"DistSQL plans didn't match:\nexpected:%s\nactual: %s",
+								testCase.expectedPlanURL, actualPlanURL,
+							)
+						}
+						return nil
+					})
 
 					errorOrigin := []int{allNodeIdxs[rng.Intn(len(allNodeIdxs))]}
 					if testCase.overrideErrorOrigin != nil {

--- a/pkg/sql/rowexec/processors_test.go
+++ b/pkg/sql/rowexec/processors_test.go
@@ -54,6 +54,7 @@ import (
 
 func TestPostProcess(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [10]rowenc.EncDatum{}
 	for i := range v {
@@ -225,6 +226,7 @@ func TestPostProcess(t *testing.T) {
 
 func TestAggregatorSpecAggregationEquals(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	// Used for FilterColIdx *uint32.
 	colIdx1 := uint32(0)
@@ -315,6 +317,7 @@ func TestAggregatorSpecAggregationEquals(t *testing.T) {
 
 func TestProcessorBaseContext(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	// Use a custom context to distinguish it from the background one.
 	ctx := context.WithValue(context.Background(), struct{}{}, struct{}{})
@@ -428,6 +431,7 @@ func populateRangeCacheAndDisableBuffering(t *testing.T, db *gosql.DB, tableName
 // interesting to test is the integration between DistSQL and KV.
 func TestDrainingProcessorSwallowsUncertaintyError(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	// We're going to test by running a query that selects rows 1..10 with limit
 	// 5. Out of these, rows 1..5 are on node 1, 6..10 on node 2. We're going to
@@ -816,6 +820,7 @@ func TestUncertaintyErrorIsReturned(t *testing.T) {
 // instantiation of processors.
 func TestFlowConcurrentTxnUse(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	t.Run("TestSingleGoroutine", func(t *testing.T) {
 		flow := &flowinfra.FlowBase{}

--- a/pkg/sql/rowexec/project_set_test.go
+++ b/pkg/sql/rowexec/project_set_test.go
@@ -28,6 +28,7 @@ import (
 
 func TestProjectSet(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [10]rowenc.EncDatum{}
 	for i := range v {

--- a/pkg/sql/rowexec/sorter_test.go
+++ b/pkg/sql/rowexec/sorter_test.go
@@ -37,6 +37,7 @@ import (
 
 func TestSorter(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	v := [6]rowenc.EncDatum{}
 	for i := range v {
@@ -339,6 +340,7 @@ func TestSorter(t *testing.T) {
 // an invalid k-parameter.
 func TestSortInvalidLimit(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	spec := execinfrapb.SorterSpec{}
 	spec.Limit = 0

--- a/pkg/sql/rowexec/stats_test.go
+++ b/pkg/sql/rowexec/stats_test.go
@@ -17,12 +17,14 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils/distsqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 )
 
 // TestInputStatCollector verifies that an inputStatCollector correctly collects
 // stats from an input.
 func TestInputStatCollector(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
 
 	const numRows = 100
 

--- a/pkg/sql/rowexec/tablereader_test.go
+++ b/pkg/sql/rowexec/tablereader_test.go
@@ -45,8 +45,9 @@ import (
 
 func TestTableReader(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	ctx := context.Background()
+	defer log.Scope(t).Close(t)
 
+	ctx := context.Background()
 	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(ctx)
 

--- a/pkg/sql/rowexec/values_test.go
+++ b/pkg/sql/rowexec/values_test.go
@@ -31,6 +31,8 @@ import (
 
 func TestValuesProcessor(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	rng, _ := randutil.NewTestRand()
 	for _, numRows := range []int{0, 1, 10, 13, 15} {
 		for _, numCols := range []int{0, 1, 3} {

--- a/pkg/sql/rowexec/windower_test.go
+++ b/pkg/sql/rowexec/windower_test.go
@@ -37,6 +37,8 @@ import (
 
 func TestWindowerAccountingForResults(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
 	ctx := context.Background()
 	st := cluster.MakeTestingClusterSettings()
 	monitor := mon.NewMonitorWithLimit(

--- a/pkg/sql/rowexec/zigzagjoiner_test.go
+++ b/pkg/sql/rowexec/zigzagjoiner_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/stretchr/testify/require"
@@ -58,8 +59,9 @@ func encInt(i int) rowenc.EncDatum {
 
 func TestZigzagJoiner(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	ctx := context.Background()
+	defer log.Scope(t).Close(t)
 
+	ctx := context.Background()
 	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(ctx)
 
@@ -741,8 +743,9 @@ func TestZigzagJoiner(t *testing.T) {
 // is closed.
 func TestZigzagJoinerDrain(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	ctx := context.Background()
+	defer log.Scope(t).Close(t)
 
+	ctx := context.Background()
 	s, sqlDB, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop(ctx)
 

